views/js/qtiCreator/widgets/interactions/gapMatchInteraction/states/Correct.js

```diff
@@ -15,44 +15,57 @@ define([
         var response = interaction.getResponseDeclaration();
 
         var corrects  = _.values(response.getCorrect());
-       
-        commonRenderer.resetResponse(interaction); 
+
+        var bodyLength = interaction.bdy.bdy.length;
+        var numberOfGaps = Object.keys(interaction.bdy.elements).length;
+        var instruction;
+
+        commonRenderer.resetResponse(interaction);
         commonRenderer.destroy(interaction);
-        
+
         //add a specific instruction
-        instructionMgr.appendInstruction(interaction, __('Please fill the gap with the correct choices.'));
-       
+        if (bodyLength === 0) { // no method?
+            instruction = __('This interaction has no text defined in question mode.');
+        }
+        else if (numberOfGaps === 0) {
+            instruction = __('Please add one or more gaps to the text in question mode.');
+        }
+        else {
```

Comment:
we don't even need this else here, instruction can have this value by default

Reply:
Could be... I like to have all the assignments to this var at the same level of indentation - neither way is bad, the way I see it.